autotest/alg/warp.py

```diff
@@ -1669,3 +1669,73 @@ def test_warp_average_oversampling():
             assert math.isnan(got_data[(y + 4) * 14 + (14 - 1 - x)])
         for x in range(6):
             assert got_data[(y + 4) * 14 + (x + 4)] == 3.0
+
+
+###############################################################################
+# Test bug fix for https://github.com/qgis/QGIS/issues/56288
+
+
+def test_non_square():
```

Comment:
to fix failure on the windows-minimal config (that removes all optional drivers like XYZ), cf https://github.com/OSGeo/gdal/actions/runs/8080871845/job/22078186296?pr=9336:

```suggestion
@pytest.mark.require_driver("XYZ")
def test_non_square():
```